baselines/diabetic_retinopathy_detection/deterministic.py

```diff
@@ -243,12 +287,14 @@ def step_fn(inputs):
     for metric in metrics.values():
       metric.reset_states()
 
-    if checkpoint_interval > 0 and (epoch + 1) % checkpoint_interval == 0:
-      checkpoint_name = checkpoint.save(os.path.join(output_dir, 'checkpoint'))
+    if (FLAGS.checkpoint_interval > 0 and
+            (epoch + 1) % FLAGS.checkpoint_interval == 0):
+      checkpoint_name = checkpoint.save(
+        os.path.join(FLAGS.output_dir, 'checkpoint'))
       logging.info('Saved checkpoint to %s', checkpoint_name)
 
   final_checkpoint_name = checkpoint.save(
```

Comment:
Below this line, you could add `model.save` followed by `model.load` during the ensemble file if you're running into any difficulties matching the compute graph in instantiating the model graph with the checkpoint's parameters. That said, these changes may take some time so if the code already works, +1 to submitting this PR as-is!

Reply:
Thanks for checking this out -- I'll try out this fix, then submit the PR later today!